Debug/Scripts/x86_64_target_definition.py

```diff
@@ -731,53 +728,50 @@ def get_reg_num(reg_num_dict, reg_name):
-            if 'slice' not in reg_info and 'composite' not in reg_info:
-                reg_info['offset'] = offset
-                offset += reg_info['bitsize'] // 8
+        # Only fill in the offset if there is no 'slice' in the register
+        # info
+        if 'slice' not in reg_info and 'composite' not in reg_info:
+            reg_info['offset'] = offset
+            offset += reg_info['bitsize'] // 8
 
-            # Set the GCC/DWARF register number for this register if it has one
-            reg_num = get_reg_num(name_to_gcc_dwarf_regnum, reg_name)
-            if reg_num != LLDB_INVALID_REGNUM:
-                reg_info['gcc'] = reg_num
-                reg_info['dwarf'] = reg_num
+        # Set the GCC/DWARF register number for this register if it has one
+        reg_num = get_reg_num(name_to_gcc_dwarf_regnum, reg_name)
+        if reg_num != LLDB_INVALID_REGNUM:
+            reg_info['gcc'] = reg_num
+            reg_info['dwarf'] = reg_num
 
-            # Set the generic register number for this register if it has one
-            reg_num = get_reg_num(name_to_generic_regnum, reg_name)
-            if reg_num != LLDB_INVALID_REGNUM:
-                reg_info['generic'] = reg_num
+        # Set the generic register number for this register if it has one
+        reg_num = get_reg_num(name_to_generic_regnum, reg_name)
+        if reg_num != LLDB_INVALID_REGNUM:
+            reg_info['generic'] = reg_num
 
-            # Set the GDB register number for this register if it has one
-            reg_num = get_reg_num(name_to_gdb_regnum, reg_name)
-            if reg_num != LLDB_INVALID_REGNUM:
-                reg_info['gdb'] = reg_num
+        # Set the GDB register number for this register if it has one
+        reg_num = get_reg_num(name_to_gdb_regnum, reg_name)
+        if reg_num != LLDB_INVALID_REGNUM:
+            reg_info['gdb'] = reg_num
 
-        g_target_definition['sets'] = [
-            'General Purpose Registers',
-            'Floating Point Registers']
-        g_target_definition['registers'] = x86_64_register_infos
-        g_target_definition[
-            'host-info'] = {'triple': triple, 'endian': eByteOrderLittle}
-        g_target_definition['g-packet-size'] = offset
+    g_target_definition['sets'] = [
+        'General Purpose Registers',
+        'Floating Point Registers']
+    g_target_definition['registers'] = x86_64_register_infos
+    g_target_definition[
+        'host-info'] = {'triple': triple, 'endian': eByteOrderLittle}
+    g_target_definition['g-packet-size'] = offset
     return g_target_definition
 
 
-def get_dynamic_setting(target, setting_name):
+def get_dynamic_setting(_, setting_name):
     if setting_name == 'gdb-server-target-definition':
         triple = os.getenv('EFI_TRIPLE')
         if not triple:
             print('Cannot find requested triple!\nHint: set EFI_TRIPLE variable.')
             sys.exit(1)
         return get_target_definition(triple)
+    return None
```

Comment:
Should not be None.

Reply:
The issue there is that prospector blames about this function not returning anything if the `if` code isn't executed. What should it return if `setting_name != 'gdb-server-target-definition'`?